python/pyarrow/tests/test_io.py

```diff
@@ -1050,3 +1055,218 @@ def test_compressed_roundtrip(compression):
     with pa.CompressedInputStream(raw, compression) as compressed:
         got = compressed.read()
         assert got == data
+
+
+# ----------------------------------------------------------------------
+# High-level API
+
+if PY2:
+    def gzip_compress(data):
+        fd, fn = tempfile.mkstemp(suffix='.gz')
+        try:
+            os.close(fd)
+            with gzip.GzipFile(fn, 'wb') as f:
+                f.write(data)
+            with open(fn, 'rb') as f:
+                return f.read()
+        finally:
+            os.unlink(fn)
+
+    def gzip_decompress(data):
+        fd, fn = tempfile.mkstemp(suffix='.gz')
+        try:
+            os.close(fd)
+            with open(fn, 'wb') as f:
+                f.write(data)
+            with gzip.GzipFile(fn, 'rb') as f:
+                return f.read()
+        finally:
+            os.unlink(fn)
+else:
+    gzip_compress = gzip.compress
+    gzip_decompress = gzip.decompress
+
+
+def test_input_stream_buffer():
+    data = b"some test data\n" * 10 + b"eof\n"
+    for arg in [pa.py_buffer(data), memoryview(data)]:
+        stream = pa.input_stream(arg)
+        assert stream.read() == data
+
+    gz_data = gzip_compress(data)
+    stream = pa.input_stream(memoryview(gz_data))
+    assert stream.read() == gz_data
+    stream = pa.input_stream(memoryview(gz_data), compression='gzip')
+    assert stream.read() == data
+
+
+def test_input_stream_file_path(tmpdir):
+    data = b"some test data\n" * 10 + b"eof\n"
+    file_path = tmpdir / 'input_stream'
+    with open(str(file_path), 'wb') as f:
+        f.write(data)
+
+    stream = pa.input_stream(file_path)
+    assert stream.read() == data
+    stream = pa.input_stream(str(file_path))
+    assert stream.read() == data
+    if pathlib is not None:
```

Comment:
Pathlib is already present in the testing suite https://github.com/apache/arrow/blob/e26806eeb4dcaa07b2f7dd7a7b3da2d0fdb42276/python/pyarrow/tests/conftest.py#L20-L23

Reply:
Ok, I see.